templates/etc/pam.d/auth.epp

```diff
@@ -0,0 +1,198 @@
+<%-|
+  String[1]                      $name,
+  Pam::PasswordBackends          $password_check_backend,
+  Optional[Stdlib::Absolutepath] $locale_file,
+  Boolean                        $cracklib_enforce_for_root,
+  Boolean                        $cracklib_reject_username,
+  Optional[Integer[0]]           $cracklib_difok,
+  Optional[Integer[0]]           $cracklib_maxrepeat,
+  Optional[Integer[0]]           $cracklib_maxsequence,
+  Optional[Integer[0]]           $cracklib_maxclassrepeat,
+  Optional[Boolean]              $cracklib_gecoscheck,
+  Optional[Integer]              $cracklib_dcredit,
+  Optional[Integer]              $cracklib_ucredit,
+  Optional[Integer]              $cracklib_lcredit,
+  Optional[Integer]              $cracklib_ocredit,
+  Optional[Integer[0]]           $cracklib_minclass,
+  Optional[Integer[0]]           $cracklib_minlen,
+  Integer[0]                     $cracklib_retry,
+  Integer[0]                     $deny,
+  Boolean                        $faillock,
+  Boolean                        $display_account_lock,
+  Integer[0]                     $fail_interval,
+  Integer[0]                     $remember,
+  Integer[0]                     $remember_retry,
+  Boolean                        $remember_for_root,
+  Boolean                        $even_deny_root,
+  Integer[0]                     $root_unlock_time,
+  Pam::HashAlgorithm             $hash_algorithm,
+  Integer[0]                     $rounds,
+  Integer[0]                     $uid,
+  Pam::AccountUnlockTime         $unlock_time,
+  Boolean                        $preserve_ac,
+  Boolean                        $use_netgroups,
+  Boolean                        $use_openshift,
+  Boolean                        $sssd,
+  Array[String[0]]               $tty_audit_users,
+  String[0]                      $separator,
+  Boolean                        $enable_separator,
+  Boolean                        $oath,
+  Integer[0]                     $oath_window
+|-%>
+#%PAM-1.0
+# This file managed by Puppet
+# User changes will be lost!
+<%
+if $even_deny_root {
+  $_even_deny_root = " even_deny_root root_unlock_time=${root_unlock_time}"
+}
+else {
+  $_even_deny_root = ''
+}
+-%>
+auth     optional      pam_faildelay.so
+auth     required      pam_env.so
+<%
+if $faillock {
+  if $display_account_lock {
+    $_display_account_lock = ''
+  }
+  else {
+    $_display_account_lock = ' silent'
+  }
+-%>
+auth     required      pam_faillock.so preauth<%= $_display_account_lock %> deny=<%= $deny %> audit unlock_time=<%= $unlock_time %> fail_interval=<%= $fail_interval %><%= $_even_deny_root %>
+<% } -%>
```

Comment:
I think this is the closing parentheses for ``if $faillock {`` but the indentation makes it difficult be be sure.